submissions/YuraZagor/dom tsk/index.js

```diff
@@ -0,0 +1,74 @@
+import { data } from './data.js'  
+
+const body = document.querySelector('body')
+let text = data[0].content
+
+const div = document.createElement('div')
+div.classList.add('wrapper')
+
+body.appendChild(div)
+
+const main = document.createElement('main')
+main.classList.add('main')
+div.appendChild(main)	
+
+const header = document.createElement('header')
+header.classList.add('header')
+main.appendChild(header)
+
+const content = document.createElement('div')
+content.classList.add('content')
+main.appendChild(content)
+
+
+const footer = document.createElement('footer')
+footer.classList.add('footer')
+main.appendChild(footer)
+
+
+content.innerHTML = text
+
+const ul = document.createElement('ul')
+ul.classList.add('sidenav--ul')
+main.insertAdjacentElement('beforebegin', ul)
+
+for (let i=0; i<data.length; i++){
+	let sidenavLi = document.createElement('li')
+	sidenavLi.classList.add('sidenav--li')
+
+	ul.appendChild(sidenavLi)
+	sidenavLi.setAttribute('id', data[i].number-1)
+	const img = document.createElement('img')
+	img.classList.add('li--img')
+	img.setAttribute('src', data[i].img )
+	img.setAttribute('alt', data[i].text )
+	sidenavLi.appendChild(img)
+}
+
+document.getElementById('0').addEventListener ('click', ()=> {
+	text= data[0].content
+	content.innerHTML = text
+	})
+document.getElementById('1').addEventListener ('click', ()=> {
+	text= data[1].content
+	content.innerHTML = text
+	})
```

Comment:
Read about [event delegation](https://javascript.info/event-delegation).

Reply:
I tried it in  some ways but it didn't work for some reason. Can it be that it is influenced by the fact the 'li' items ul-listener adresses aren't generated yet when I run the code?

Reply:
found the bug in my code, I had to declare a variable outside function scope to make it work in clickhandler function, easy as that

Thanks for the link to delegation topic